project-base/storefront/components/Basic/GoogleMap/GoogleMap.tsx

```diff
@@ -0,0 +1,142 @@
+import { GoogleMapMarker } from './GoogleMapMarker';
+import { useDomainConfig } from 'components/providers/DomainConfigProvider';
+import GoogleMapReact from 'google-map-react';
+import getConfig from 'next/config';
+import { useEffect, useMemo, useState } from 'react';
+import useSupercluster from 'use-supercluster';
+
+type GoogleMapMarker = {
+    identifier?: string;
+    uuid?: string;
+    locationLatitude: string | null;
+    locationLongitude: string | null;
+};
+
+type GoogleMapProps = {
+    lat?: string | null;
+    lng?: string | null;
+    zoom?: number | null;
+    markers?: GoogleMapMarker[];
+    activeMarkerHandler?: (id: string) => void;
+    isDetail?: boolean;
+    closeMarkers?: boolean;
+};
+
+const TEST_IDENTIFIER = 'basic-googlemap';
+
+const ClusterMarker: FC<{
+    lat: number;
+    lng: number;
+}> = ({ children }) => (
+    <div className="absolute -translate-x-1/2 -translate-y-1/2">
+        <div className="flex justify-center items-center w-12 h-12 bg-white border-8 border-transparent rounded-full font-bold">
+            {children}
+        </div>
+    </div>
+);
+
+export const GoogleMap: FC<GoogleMapProps> = ({
+    lat,
+    lng,
+    zoom: zoomProps,
+    markers,
+    activeMarkerHandler,
+    isDetail,
+    closeMarkers,
+}) => {
+    const { publicRuntimeConfig } = getConfig();
+    const { mapSetting } = useDomainConfig();
+    const defaultLat = lat === null || lat === undefined ? mapSetting.latitude : parseFloat(lat);
+    const defaultLng = lng === null || lng === undefined ? mapSetting.longitude : parseFloat(lng);
+    const defaultZoom = zoomProps === null || zoomProps === undefined ? mapSetting.zoom : zoomProps;
+    const [activeMarkerId, setActiveMarkerId] = useState('');
+    const [zoom, setZoom] = useState(defaultZoom);
+    const [bounds, setBounds] = useState<number[]>([]);
+
+    const points = useMemo(
```

Comment:
Taking _markers_ and mapping it into _points_ might be confusing. Is it necessary to use a completely different name, instead of e.g. _markersClusterConfig_ or similar?

Reply:
Yes i agree.